azurerm/internal/services/datafactory/data_factory_linked_service_azure_blob_storage_resource.go

```diff
@@ -202,6 +269,17 @@ func resourceDataFactoryLinkedServiceBlobStorageRead(d *schema.ResourceData, met
 		return fmt.Errorf("Error classifiying Data Factory Linked Service BlobStorage %q (Data Factory %q / Resource Group %q): Expected: %q Received: %q", name, dataFactoryName, resourceGroup, datafactory.TypeAzureBlobStorage, *resp.Type)
 	}
 
+	if blobStorage.Tenant != nil {
+		d.Set("tenant", blobStorage.Tenant)
+	}
+
+	if blobStorage.ServicePrincipalID != nil {
+		d.Set("service_principal_id", blobStorage.ServicePrincipalID)
+		d.Set("use_managed_identity", false)
+	} else {
+		d.Set("use_managed_identity", true)
+	}
+
```

Comment:
We'll need to nil check `blobStorage.AzureBlobStorageLinkedServiceTypeProperties` before trying to access `Tenant` / `ServicePrincipalID` etc or these could cause crashes.